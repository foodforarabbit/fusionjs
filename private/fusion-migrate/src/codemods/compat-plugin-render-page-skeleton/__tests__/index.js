const path = require('path');
const pluginTester = require('../../../utils/codemod-test.js');
const babylon = require('babylon');

const fixtureDir = path.join(__dirname, '../__fixtures__');
const plugin = require('../plugin');

pluginTester(
  fixtureDir,
  plugin({pageSkeletonConfig: babylon.parseExpression(`{useMagellan: true}`)})
);
