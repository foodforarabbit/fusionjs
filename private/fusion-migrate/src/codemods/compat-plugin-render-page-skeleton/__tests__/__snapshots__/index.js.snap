// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compat-plugin-render-page-skeleton positive-match: positive-match 1`] = `
"import App from 'fusion-react';
import {createPlugin} from 'fusion-core';
import {SomeToken, SomePlugin} from 'some-plugin';
import {OtherToken, OtherPlugin} from 'other-plugin';

console.log(App);

function thing() {
  const app = new App();
  if (__BROWSER__) {
    app.register(OtherToken, OtherPlugin);
  }
  if (__NODE__) {
    app.register(SomeToken, SomePlugin);
  }
  return app;
}


      ↓ ↓ ↓ ↓ ↓ ↓

import App from 'fusion-react';
import {createPlugin, RenderToken} from 'fusion-core';
import PageSkeletonRenderer, {
  PageSkeletonConfigToken,
} from '@uber/fusion-plugin-page-skeleton-compat';
import {SomeToken, SomePlugin} from 'some-plugin';
import {OtherToken, OtherPlugin} from 'other-plugin';
console.log(App);

function thing() {
  const app = new App();

  if (__BROWSER__) {
    app.register(OtherToken, OtherPlugin);
  }

  if (__NODE__) {
    app.register(SomeToken, SomePlugin);
    app.register(RenderToken, PageSkeletonRenderer);
    app.register(PageSkeletonConfigToken, {
      useMagellan: true,
    });
  }

  return app;
}
"
`;
